lib/gush/client.rb

```diff
@@ -83,13 +94,18 @@ def persist_workflow(workflow)
     end
 
     def persist_job(workflow_id, job)
-      redis.set("gush.jobs.#{workflow_id}.#{job.class.to_s}", job.to_json)
+      redis.set("gush.jobs.#{workflow_id}.#{job.name}", job.to_json)
     end
 
     def load_job(workflow_id, job_id)
       workflow = find_workflow(workflow_id)
-      data = redis.get("gush.jobs.#{workflow_id}.#{job_id}")
-      return nil if data.nil?
+
+      keys = redis.keys("gush.jobs.#{workflow_id}.#{job_id}*")
```

Comment:
Would be good to have some sort of separator between `#{job_id}` and `*` so we don't catch jobs starting with the same prefix. I think a regular hyphen will suffice :)

Example:

If I have two jobs named `FetchSomething` and `FetchSomethingElse` then loading `FetchSomething` will also find the latter.